src/auth/oauth2client.ts

```diff
@@ -216,14 +216,20 @@ export interface VerifyIdTokenOptions {
   maxExpiry?: number;
 }
 
-export interface OAuth2ClientOptions {
+export interface OAuth2ClientOptions extends RefreshOptions {
   clientId?: string;
   clientSecret?: string;
   redirectUri?: string;
   authBaseUrl?: string;
   tokenUrl?: string;
 }
 
+export interface RefreshOptions {
+  // The token should be refreshed if it will expire within this many
+  // milliseconds.
```

Comment:
Use jsdoc style comments so that this gets emitted into the docs.

nit: `refreshTokenEarlyMillis` doesn't seem obviously descriptive to me. Perhaps you can name it 'eagerRefreshThresholdMillis'?

I would suggest phrasing the comment as something like: "Eagerly refresh unexpired tokens when they are within this many milliseconds from expiring".

Reply:
nit: the default value should be documented. Right now the default value is buried below in the initialization in the constructor. Is 0 a reasonable default value? /cc @JustinBeckwith

Reply:
SGTM

Reply:
Given the issues raised in #62 I would rather see this defaulted to at least 1 second.

Reply:
Renamed, updated the comment, and increased the default value 1 second.